src/ResourceManager/Profile/Commands.Profile/Subscription/GetAzureRMSubscription.cs

```diff
@@ -92,7 +95,41 @@ public override void ExecuteCmdlet()
             {
                 try
                 {
-                    WriteObject(_client.GetSubscriptions(tenant).Select((s) => (PSAzureSubscription)s), enumerateCollection: true);
+                    var tenantsList = new List<string>();
+
+                    if (string.IsNullOrWhiteSpace(tenant))
+                    {
+                        tenantsList.AddRange(_client.ListTenants()
+                            .Select(t => (t.Id == Guid.Empty) ? t.Domain : t.Id.ToString()));
+                    }
+                    else
+                    {
+                        tenantsList.Add(tenant);
+                    }
+
+                    foreach (var tenantId in tenantsList)
+                    {
+                        try
+                        {
+                            string listNextLink = null;
+                            do
+                            {
+                                var subscriptions = _client.ListSubscriptions(tenantId, ref listNextLink);
+                                WriteObject(subscriptions.Select((s) => (PSAzureSubscription)s), enumerateCollection: true);
```

Comment:
@hovsepm I'm not sure what actually happens in this case, where we are writing mulsiple collections to the output stream.  To be safe, I would perform a WriteObject for each subscription.